os/include/tinyara/testcase_drv.h

```diff
@@ -52,7 +52,26 @@
  *
  */
 
-#define TESTIOC_DRIVER_ANALOG            _TESTIOC(1)
+#define TESTIOC_ANALOG                          _TESTIOC(1)
+#define TESTIOC_CLOCK_ABSTIME2TICKS             _TESTIOC(2)
+#define TESTIOC_GET_SELF_TCB_PID                _TESTIOC(3)
+#define TESTIOC_GET_SIG_FINDACTION_ADD          _TESTIOC(4)
+#define TESTIOC_GET_TCB_ALIVE                   _TESTIOC(5)
+#define TESTIOC_GET_TCB_LOCKCOUNT               _TESTIOC(6)
+#define TESTIOC_GET_TCB_PID                     _TESTIOC(7)
+#define TESTIOC_GET_TCB_SIGPROCMASK             _TESTIOC(8)
+#define TESTIOC_GET_TCB_STACK_SIZE              _TESTIOC(9)
+#define TESTIOC_GET_TCB_TIMESLICE               _TESTIOC(10)
+#define TESTIOC_SCHED_FOREACH                   _TESTIOC(11)
+#define TESTIOC_TC_GROUP_ADD_FIND_REMOVE_CHILD  _TESTIOC(12)
+#define TESTIOC_TC_GROUP_ALLOC_FREE_CHILD       _TESTIOC(13)
+#define TESTIOC_TC_GROUP_EXIT_CHILD             _TESTIOC(14)
+#define TESTIOC_TC_GROUP_REMOVE_CHILDREN        _TESTIOC(15)
+#define TESTIOC_TC_WORKQUEUE                    _TESTIOC(16)
+#define TESTIOC_TC_SEM_TICKWAIT                 _TESTIOC(17)
+#define TESTIOC_TIMER_INITIALIZE                _TESTIOC(18)
```

Comment:
I see that you are adding one TESTIOC_xxxxxx for each test case from the user space. 
However, the max range of TESTIOC_xxxx is from 0x00 to 0xFF only. Hence we can add max 255 TESTIOC_xxx types. 

Our original idea for Testcase driver was to add one TESTIOC_xxxx for each module. For example, Scheduler will have TESTIOC_SCHED, timer will have TESTIOC_TIMER and so on. Individual testcases within a module will be indexed using the arg parameter sent to the ioctl. For example we have the below definitions for testcases in scheduler module.

#define TESTIOC_TC_SCHED                    _TESTIOC(16)


#define TC_SCHED_TCB_PID  0x00
#define TC_SCHED_TCB_TIMESLICE  0x01
...
...
etc...

Please consider changing your implementation to this method, so that we can accomodate large number of testcases from all modules.

Reply:
@YiHyunJin Could you give me your opinion for @kishore-sn 's suggestion?

Reply:
@sunghan-chang yes. I changed.